python/revng/internal/cli/_commands/mass_testing/configure.py

```diff
@@ -0,0 +1,70 @@
+#
+# This file is distributed under the MIT License. See LICENSE.md for details.
+#
+
+import argparse
+import os
+from pathlib import Path
+from random import shuffle
+from shutil import copy2
+from tempfile import NamedTemporaryFile
+
+import yaml
+
+from revng.internal.cli.support import run
+
+from ...commands_registry import Command, Options
+
+
+class MassTestingConfigureCommand(Command):
+    def __init__(self):
+        super().__init__(
+            ("mass-testing", "configure"), "Collect binary files and run test-configure"
+        )
+
+    def register_arguments(self, parser: argparse.ArgumentParser):
+        parser.description = "Collect binary files and run test-configure"
+        parser.add_argument("--meta", help="Metadata definition")
+        parser.add_argument("input", help="Input directory")
+        parser.add_argument("output", help="Output directory")
+        parser.add_argument(
+            "config_files", nargs="+", help="Additional .yml files to pass to test-configure"
+        )
+
+    def run(self, options: Options):
+        args = options.parsed_args
+
+        file_list = []
+        input_path = Path(args.input)
+        for dirpath, _, filenames in os.walk(args.input):
+            dirpath_path = Path(dirpath)
+            for filename in filenames:
+                file_list.append(str((dirpath_path / filename).relative_to(input_path)))
+
+        # Shuffle the input files, this avoids having a limited variety of
+        # test runs for a partial run
```

Comment:
Basically, modulo filesystem ordering (which is usually sorted), the order of execution of the individual "tests" will be lexicographic, e.g.

```
bar/file_0001
bar/file_0002
...
bar/file_1000
foo/file_0001
foo/file_0002
...
foo/file_1000
```

If we don't `shuffle` that means that a partial run (e.g. where we ctrl-c mid-way) might have (in this example) an over-representation of `bar` and few or no members of `foo`.

Reply:
Mmmh, what's the use case to Ctrl+C midway?

This reduces determinism, if there's a bug, it will reproduce at different latencies, which can be annoying.
I'm OK with the idea of shuffling, but let's make it deterministic. Get some shuffling routine which enables you do set the seed, so it's "random" but predictably so.

Reply:
> Mmmh, what's the use case to Ctrl+C midway?

A couple that come to mind:
* Mass testing takes too long, we get impatient and want to see results
* The test container/machine crashes, we want to salvage the work that has been done

> This reduces determinism, if there's a bug, it will reproduce at different latencies, which can be annoying.

This basically only effects the order in which `ninja` runs things, I doubt there's a bug that's inter-job.

Reply:
Implemented